code/model/SiteTree.php

```diff
@@ -1735,43 +1710,42 @@ public function getLiveURLSegment() {
 	}
 	
 	/**
-	 * Rewrites any linked images on this page.
+	 * Rewrites any linked images on this page without creating a new version record.
 	 * Non-image files should be linked via shortcodes
 	 * Triggers the onRenameLinkedAsset action on extensions.
-	 * TODO: This doesn't work for HTMLText fields on other tables.
+	 *
+	 * @todo Implement image shortcodes and remove this feature
 	 */
 	public function rewriteFileLinks() {
+		// Skip live stage
+		if(\Versioned::current_stage() === \Versioned::get_live_stage()) {
+			return;
+		}
+
 		// Update the content without actually creating a new version
-		foreach(array("SiteTree_Live", "SiteTree") as $table) {
-			// Published site
-			$published = DB::prepared_query(
-				"SELECT * FROM  \"$table\" WHERE \"ID\" = ?",
-				array($this->ID)
-			)->record();
-			$origPublished = $published;
-
-			foreach($this->db() as $fieldName => $fieldType) {
-				// Skip if non HTML or if empty
-				if ($fieldType !== 'HTMLText' || empty($published[$fieldName])) {
-					continue;
-				}
+		foreach($this->db() as $fieldName => $fieldType) {
+			// Skip if non HTML or if empty
+			if ($fieldType !== 'HTMLText') {
+				continue;
+			}
+			$fieldValue = $this->{$fieldName};
+			if(empty($fieldValue)) {
+				continue;
+			}
 
-				// Regenerate content
-				$content = Image::regenerate_html_links($published[$fieldName]);
-				if($content === $published[$fieldName]) {
-					continue;
-				}
+			// Regenerate content
+			$content = Image::regenerate_html_links($fieldValue);
+			if($content === $fieldValue) {
+				continue;
+			}
 
-				$query = sprintf('UPDATE "%s" SET "%s" = ? WHERE "ID" = ?', $table, $fieldName);
-				DB::prepared_query($query, array($content, $this->ID));
+			// Write content directly without updating linked assets
+			$table = ClassInfo::table_for_object_field($this, $fieldName);
+			$query = sprintf('UPDATE "%s" SET "%s" = ? WHERE "ID" = ?', $table, $fieldName);
+			DB::prepared_query($query, array($content, $this->ID));
 
-				// Tell static caching to update itself
-				if($table == 'SiteTree_Live') {
-					$publishedClass = $origPublished['ClassName'];
-					$origPublishedObj = new $publishedClass($origPublished);
-					$this->invokeWithExtensions('onRenameLinkedAsset', $origPublishedObj);
-				}
-			}
+			// Update linked assets
+			$this->invokeWithExtensions('onRenameLinkedAsset');
```

Comment:
> $origPublishedObj

Not so much not needed as no longer available. I checked the static publisher module and it doesn't use that parameter anyway.

This method will get nuked in a near update in any case; once all images are shortcoded we won't need to do this anymore.

I expect that static publisher will need to hook into another extension point to publish pages; onAfterPublish etc.